script.js

```diff
@@ -1,12 +1,169 @@
+    .catch(function (error) {
+      console.log(error);
+    });
+}
+
+function tvSeriesSetup(showList) {
+  let selectSeries = document.getElementById("show-select");
+  let optionSeries = document.createElement("option");
+  optionSeries.textContent = "Please Choose Your Show";
+  selectSeries.appendChild(optionSeries);
+  selectSeries.selectedIndex = 0;
+
+  //sorting based on alphabetical order
+  showList.sort(function (a, b) {
+    return a.name.localeCompare(b.name);
+  });
+  for (let i = 0; i < showList.length; i++) {
+    let option = document.createElement("option");
+    option.value = showList[i].id;
+    option.textContent = `${showList[i].name}`;
+    selectSeries.appendChild(option);
+  }
+  selectSeries.addEventListener("change", function () {
+    console.log(selectSeries.value);
+    fetchAllEpisodes(selectSeries.value);
+  });
+}
+//its optional will refactor later
+function setup() {
+  const seriesList = getAllShows();
+  tvSeriesSetup(seriesList);
+}
+//creating search bar//
+const searchElem = document.querySelector("#search");
+searchElem.addEventListener("input", searchEpisode);
+function searchEpisode() {
+  const searchInput = searchElem.value.toLowerCase();
+  const filteredEpisodes = allEpisodes.filter((episode) => {
+    if (
+      episode.name.toLowerCase().includes(searchInput) ||
+      episode.summary.toLowerCase().includes(searchInput)
+    ) {
+      return episode;
+    }
+  });
+
+  document.querySelector("#num").innerText = filteredEpisodes.length;
+  makePageForEpisodes(filteredEpisodes);
+}
+//creating function//
 function makePageForEpisodes(episodeList) {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
-}
+  rootElem.textContent = "";
+
+  //top container which will wrap all movieCart
+  const topContainer = document.createElement("div");
+  topContainer.classList.add("top-container");
+  rootElem.appendChild(topContainer);
+  for (let i = 0; i < episodeList.length; i++) {
+    //this is movieCard in which all title, photo and summery will be appended.
+    const movieCard = document.createElement("div");
+    movieCard.classList.add("movie-card");
+    const testElement = document.createElement("p");
+    movieCard.appendChild(testElement);
+    topContainer.appendChild(movieCard);
+
+    //Assigning TV episodes in p element
+    let episodeName = document.createElement("p");
+    episodeName.classList.add("epName");
+    movieCard.appendChild(episodeName);
 
+    //assigning the season number & episode number//
+    episodeSeason = episodeList[i].season;
+    episodeNumber = episodeList[i].number;
+    if (episodeSeason < 10) {
+      episodeSeason = "0" + episodeSeason;
+    }
+    if (episodeNumber < 10) {
+      episodeNumber = "0" + episodeNumber;
+    }
+    episodeName.innerText =
+      episodeList[i].name +
+      " " +
+      "-" +
+      " " +
+      "S" +
+      episodeSeason +
+      "E" +
+      episodeNumber;
```

Comment:
It would be more preferrable and easier to read if you use template literals here.

Reply:
Here i just replace with:
`${episodeList[i].name} - S${episodeSeason}E${episodeNumber}`
Thank you